### PR TITLE
Fix hotplug volume have a io error when use iscsi protocol csi plugin

### LINKED ISSUE #6564: ADD hotplug volume more than one will cause io error
**What happened**:
when I first add one hotplug volume to a vm ,everything is ok.
but when I second add one or more volume to a vm ,hotplug volume was added before will appear the io error.
**What you expected to happen**:
vm will be paused , because the io error
**How to reproduce it (as minimally and precisely as possible)**:
you can try add hotplug volume in the different time to a vm.

**Anything else we need to know?**:
hotplug volume is use block mode

**Environment**:
- KubeVirt version (use `0.44.0`):

### PR DESCRIPTION
**Pull request by watermelon-brother**
**What this PR does / why we need it**:
Now, virt-controller can delete old attachment pod when the new attachmentpod is running.So,It can't cause the volume  iscsi connection break when delete the old attachment pod.
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #6564 

**Special notes for your reviewer**:

**Release note**:
None

### CODE REVIEW

**Comment by maya-r** [comment]
fyi @awels

**Comment by watermelon-brother** [comment]
The tide stage of pipeline is not pass,How to solve it? @alicefr @maya-r

**Comment by alicefr** [comment]
@watermelon-brother you need approval and a lgtm from reviewers to get this merged.

@awels do we want this solution? Should we try it with another CSI plugin that do not accept multiple pods accessing a RWO volume?

File: pkg/virt-controller/watch/vmi.go [hunk 1]
```diff
@@ -1612,19 +1611,33 @@ func (c *VMIController) handleHotplugVolumes(hotplugVolumes []*virtv1.Volume, ho
 		}
 	}
 
-	if len(currentPod) == 0 && len(readyHotplugVolumes) > 0 {
-		// ready volumes have changed
-		// Create new attachment pod that holds all the ready volumes
-		if err := c.createAttachmentPod(vmi, virtLauncherPod, readyHotplugVolumes); err != nil {
-			return err
+	switch len(currentPod) {
```
Comment (awels) [inline pkg/virt-controller/watch/vmi.go:1614]: Any reason you used a switch instead of if len(currentPod) == 0 { ... } else { ... }

**Comment by awels** [comment]
The solution I think is fine, it is just I am curious about the switch instead of an if else since there are only 2 options.

**Comment by DMajrekar** [comment]
I've been testing this change with OpenEBS/Mayastor backed hot plugged volumes, and this PR introduces and undesired side effect. 

I've been testing with both block and file backed volumes, and executed the following steps:

1. Create a VM
2. Attach a first hot plug device `block-test`
3. Partition / Format the device within the VM: sda
4. Attach a second hot plug device `block-test-2`
5. Partition / Format the device within the VM: sdb
6. unmount the first device, sda from the VM
7. Hot plug remove the device `block-test` from the VM (`virtctl removevolume volume-attach --volume-name block-test`
8. Check the mount points within the VM, and notice that sda has been replaced with sdb, leaving sdb in an I/O error state

Looking at the virt-launcher logs for this VM, I'm seeing:

```
{"component":"virt-launcher","kind":"","level":"info","msg":"Attaching disk block-test, target sda","name":"volume-attach-2438-4bcde4","namespace":"cust-ba6bc6b1-808d-default","pos":"manager.go:929","timestamp":"2022-03-28T18:12:50.396629Z","uid":"ba639b80-983a-4304-b491-fdbc27d449bb"}
{"component":"virt-launcher","kind":"","level":"info","msg":"Attaching disk block-test-2, target sdb","name":"volume-attach-2438-4bcde4","namespace":"cust-ba6bc6b1-808d-default","pos":"manager.go:929","timestamp":"2022-03-28T18:13:16.131228Z","uid":"ba639b80-983a-4304-b491-fdbc27d449bb"}


{"component":"virt-launcher","kind":"","level":"info","msg":"Detaching disk block-test, target sda","name":"volume-attach-2438-4bcde4","namespace":"cust-ba6bc6b1-808d-default","pos":"manager.go:908","timestamp":"2022-03-28T18:13:48.450685Z","uid":"ba639b80-983a-4304-b491-fdbc27d449bb"}
{"component":"virt-launcher","kind":"","level":"info","msg":"Detaching disk block-test-2, target sdb","name":"volume-attach-2438-4bcde4","namespace":"cust-ba6bc6b1-808d-default","pos":"manager.go:908","timestamp":"2022-03-28T18:13:48.567696Z","uid":"ba639b80-983a-4304-b491-fdbc27d449bb"}
{"component":"virt-launcher","kind":"","level":"info","msg":"Attaching disk block-test-2, target sdb","name":"volume-attach-2438-4bcde4","namespace":"cust-ba6bc6b1-808d-default","pos":"manager.go:929","timestamp":"2022-03-28T18:13:48.747398Z","uid":"ba639b80-983a-4304-b491-fdbc27d449bb"}
```

I however have been experiencing the same problems as seen in #6564 with OpenEBS/Mayastor which relies on NVMe over fabric rather than iSCSI. 

Working though the issue, we have identified that the 2nd and 3rd hot plug pods are missing the `volumeDevices` section for previously mounted volumes. That is causing the Node CSI driver to performa an UNSTAGE of the volume (effectively removing the block device from the underlying compute node). 

As a quick hack, I have forced subsequent hot plug pods to maintain the `volumeDevice` spec with the following patch

```
diff --git a/pkg/virt-controller/services/template.go b/pkg/virt-controller/services/template.go
index b1e102efa..892598387 100644
--- a/pkg/virt-controller/services/template.go
+++ b/pkg/virt-controller/services/template.go
@@ -1622,7 +1622,7 @@ func (t *templateService) RenderHotplugAttachmentPodTemplate(volumes []*v1.Volum
                }
                skipMount := false
                if hotplugVolumeStatusMap[volume.Name] == v1.VolumeReady || hotplugVolumeStatusMap[volume.Name] == v1.HotplugVolumeMounted {
-                       skipMount = true
+                       skipMount = false
                }
                pod.Spec.Volumes = append(pod.Spec.Volumes, k8sv1.Volume{
                        Name: volume.Name,
``` 

This now means the hot plug path is UNPUBLISHED, but is _not_ UNSTAGED from the underlying compute node, meaning IO can still flow. It also does not show the same behaviour as I have described at the start of this, where the guest VM sees both devices detached and then a single device reattached. 

I will write this up in a new issue and PR tomorrow, but at the moment, I feel this PR is not safe to merge.

**Pull request closed without merging**